--- pkg/controller/direct/sql/mapper.go
+++ pkg/controller/direct/sql/mapper.go
@@ -71,28 +71,28 @@
 	return &out
 }
 
-func DatabaseInstance_PoolNodeConfigObservedState_State_ToProto(mapCtx *direct.MapContext, in *string) *pb.DatabaseInstance_PoolNodeConfig_State {
+func DatabaseInstance_PoolNodeConfigObservedState_State_ToProto(mapCtx *direct.MapContext, in *string) *pb.DatabaseInstance_SqlInstanceState {
 	if in == nil {
 		return nil
 	}
-	out := direct.Enum_ToProto[pb.DatabaseInstance_PoolNodeConfig_State](mapCtx, in)
+	out := direct.Enum_ToProto[pb.DatabaseInstance_SqlInstanceState](mapCtx, in)
 	return &out
 }
 
-func IPConfiguration_ServerCaMode_ToProto(mapCtx *direct.MapContext, in *string) *pb.IPConfiguration_ServerCaMode {
+func IPConfiguration_ServerCaMode_ToProto(mapCtx *direct.MapContext, in *string) *pb.IpConfiguration_CaMode {
 	if in == nil {
 		return nil
 	}
-	out := direct.Enum_ToProto[pb.IPConfiguration_ServerCaMode](mapCtx, in)
+	out := direct.Enum_ToProto[pb.IpConfiguration_CaMode](mapCtx, in)
 	return &out
 }
 
-func IPConfiguration_ServerCertificateRotationMode_ToProto(mapCtx *direct.MapContext, in *string) *pb.IPConfiguration_ServerCertificateRotationMode {
+func IPConfiguration_ServerCertificateRotationMode_ToProto(mapCtx *direct.MapContext, in *string) *pb.IpConfiguration_ServerCertificateRotationMode {
 	if in == nil {
 		return nil
 	}
-	out := direct.Enum_ToProto[pb.IPConfiguration_ServerCertificateRotationMode](mapCtx, in)
+	out := direct.Enum_ToProto[pb.IpConfiguration_ServerCertificateRotationMode](mapCtx, in)
 	return &out
 }
 
