diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudidentity/resource_cloud_identity_group.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudidentity/resource_cloud_identity_group.go
index b52fbde5e..ba6c7dd11 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudidentity/resource_cloud_identity_group.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudidentity/resource_cloud_identity_group.go
@@ -22,6 +22,7 @@ import (
 	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"google.golang.org/api/googleapi"
 
 	"github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource"
 	transport_tpg "github.com/hashicorp/terraform-provider-google-beta/google-beta/transport"
@@ -317,6 +318,11 @@ func resourceCloudIdentityGroupRead(d *schema.ResourceData, meta interface{}) er
 		UserAgent: userAgent,
 	})
 	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 403 && strings.Contains(gerr.Message, "Permission denied") {
+			// Deleted or uncreated Groups will always return 403 on GET
+			d.SetId("")
+			return nil
+		}
 		return transport_tpg.HandleNotFoundError(err, d, fmt.Sprintf("CloudIdentityGroup %q", d.Id()))
 	}
 
