diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_reservation.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_reservation.go
index dbeb0e5ea..5987a9e09 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_reservation.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_reservation.go
@@ -172,7 +172,6 @@ for information on available CPU platforms.`,
 			},
 			"share_settings": {
 				Type:        schema.TypeList,
-				Computed:    true,
 				Optional:    true,
 				Description: `The share setting for reservations.`,
 				MaxItems:    1,
