diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_managed_ssl_certificate.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_managed_ssl_certificate.go
index 4a722566d..6aeaa228e 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_managed_ssl_certificate.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_managed_ssl_certificate.go
@@ -103,7 +103,6 @@ which type this is. Default value: "MANAGED" Possible values: ["MANAGED"]`,
 			"certificate_id": {
 				Type:        schema.TypeInt,
 				Computed:    true,
-				Optional:    true,
 				Description: `The unique identifier for the resource.`,
 			},
 			"creation_timestamp": {
