diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_security_policy.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_security_policy.go
index 4dcd779a0..ac787ba9c 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_security_policy.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_security_policy.go
@@ -436,7 +436,7 @@ func resourceComputeSecurityPolicy() *schema.Resource {
 				MaxItems:    1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"layer_7_ddos_defense_config": {
+						"layer7_ddos_defense_config": {
 							Type:        schema.TypeList,
 							Description: `Layer 7 DDoS Defense Config of this security policy`,
 							Optional:    true,
@@ -1096,7 +1096,7 @@ func expandSecurityPolicyAdaptiveProtectionConfig(configured []interface{}) *com
 
 	data := configured[0].(map[string]interface{})
 	return &compute.SecurityPolicyAdaptiveProtectionConfig{
-		Layer7DdosDefenseConfig: expandLayer7DdosDefenseConfig(data["layer_7_ddos_defense_config"].([]interface{})),
+		Layer7DdosDefenseConfig: expandLayer7DdosDefenseConfig(data["layer7_ddos_defense_config"].([]interface{})),
 		AutoDeployConfig:        expandAutoDeployConfig(data["auto_deploy_config"].([]interface{})),
 	}
 }
@@ -1134,8 +1134,8 @@ func flattenSecurityPolicyAdaptiveProtectionConfig(conf *compute.SecurityPolicyA
 	}
 
 	data := map[string]interface{}{
-		"layer_7_ddos_defense_config": flattenLayer7DdosDefenseConfig(conf.Layer7DdosDefenseConfig),
-		"auto_deploy_config":          flattenAutoDeployConfig(conf.AutoDeployConfig),
+		"layer7_ddos_defense_config": flattenLayer7DdosDefenseConfig(conf.Layer7DdosDefenseConfig),
+		"auto_deploy_config":         flattenAutoDeployConfig(conf.AutoDeployConfig),
 	}
 
 	return []map[string]interface{}{data}
diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_security_policy_test.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_security_policy_test.go
index b00e79d2e..8bcbc115e 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_security_policy_test.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_compute_security_policy_test.go
@@ -971,7 +971,7 @@ resource "google_compute_security_policy" "policy" {
   description = "updated description"
 
   adaptive_protection_config {
-    layer_7_ddos_defense_config {
+    layer7_ddos_defense_config {
       enable = true
       rule_visibility = "STANDARD"
 	}
