diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_resource_policy.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_resource_policy.go
index 002152b0b..dc0a4f05e 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_resource_policy.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_resource_policy.go
@@ -359,7 +359,6 @@ with RFC1035.`,
 										ForceNew: true,
 										Description: `Cloud Storage bucket location to store the auto snapshot
 (regional or multi-regional)`,
-										MaxItems: 1,
 										Elem: &schema.Schema{
 											Type: schema.TypeString,
 										},
