diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_forwarding_rule.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_forwarding_rule.go
index 6ff824197..9e6573b5c 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_forwarding_rule.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute/resource_compute_forwarding_rule.go
@@ -326,8 +326,6 @@ This field is not applicable to global forwarding rules.`,
 				Description: `Service Directory resources to register this forwarding rule with.
 
 Currently, only supports a single Service Directory resource.`,
-				MinItems: 0,
-				MaxItems: 1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"namespace": {
