diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dns/resource_dns_record_set.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dns/resource_dns_record_set.go
index 84a70e465..f55b60721 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dns/resource_dns_record_set.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dns/resource_dns_record_set.go
@@ -92,13 +92,12 @@ func resourceDnsRecordSet() *schema.Resource {
 
 			"rrdatas": {
 				Type:     schema.TypeList,
-				Optional: true,
+				Required: true,
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
 				DiffSuppressFunc: rrdatasDnsDiffSuppress,
 				Description:      `The string data for the records in this record set whose meaning depends on the DNS type. For TXT record, if the string data contains spaces, add surrounding \" if you don't want your string to get split on spaces. To specify a single record value longer than 255 characters such as a TXT record for DKIM, add \"\" inside the Terraform configuration string (e.g. "first255characters\"\"morecharacters").`,
-				ExactlyOneOf:     []string{"rrdatas", "routing_policy"},
 			},
 
 			"routing_policy": {
@@ -154,7 +153,6 @@ func resourceDnsRecordSet() *schema.Resource {
 						},
 					},
 				},
-				ExactlyOneOf: []string{"rrdatas", "routing_policy"},
 			},
 
 			"ttl": {
