diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_folder.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_folder.go
index f67b5d2c8..d8ad0e4f5 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_folder.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_folder.go
@@ -2,6 +2,7 @@ package google
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"log"
 	"strings"
@@ -11,6 +12,8 @@ import (
 	resourceManagerV3 "google.golang.org/api/cloudresourcemanager/v3"
 )
 
+var activeFolderNotFoundError = errors.New("active folder not found")
+
 func ResourceGoogleFolder() *schema.Resource {
 	return &schema.Resource{
 		Create: resourceGoogleFolderCreate,
@@ -86,6 +89,19 @@ func resourceGoogleFolderCreate(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error getting parent for folder '%s': %s", displayName, err)
 	}
 
+	// Check if there's an ACTIVE folder with the given display_name in the
+	// given parent first before trying to create a new folder. This allows
+	// users to acquire existing folders by specifying the folder's
+	// display_name and parent.
+	folder, err := getActiveFolderByDisplayName(displayName, parent, userAgent, config)
+	if err != nil && !errors.Is(err, activeFolderNotFoundError) {
+		return fmt.Errorf("Error checking if folder '%s' in '%s' exists: %s", displayName, parent, err)
+	} else if err == nil {
+		// An ACTIVE folder with the given display_name in the given parent is found.
+		d.SetId(folder.Name)
+		return resourceGoogleFolderRead(d, meta)
+	}
+
 	var op *resourceManagerV3.Operation
 	err = retryTimeDuration(func() error {
 		var reqErr error
@@ -295,6 +311,27 @@ func getGoogleFolder(folderName, userAgent string, d *schema.ResourceData, confi
 	return folder, nil
 }
 
+func getActiveFolderByDisplayName(displayName, parent, userAgent string, config *Config) (*resourceManagerV3.Folder, error) {
+	pageToken := ""
+	for ok := true; ok; ok = pageToken != "" {
+		query := fmt.Sprintf("state=ACTIVE AND parent=%s AND displayName=\"%s\"", parent, displayName)
+		searchResponse, err := config.NewResourceManagerV3Client(userAgent).Folders.Search().Query(query).PageToken(pageToken).Do()
+		if err != nil {
+			if isGoogleApiErrorWithCode(err, 404) {
+				return nil, activeFolderNotFoundError
+			}
+			return nil, fmt.Errorf("error searching for folders with query '%v': %v", query, err)
+		}
+		for _, folder := range searchResponse.Folders {
+			if folder.DisplayName == displayName {
+				return folder, nil
+			}
+		}
+		pageToken = searchResponse.NextPageToken
+	}
+	return nil, activeFolderNotFoundError
+}
+
 func getParentID(d *schema.ResourceData) (string, error) {
 	orgId := d.Get("parent_org_id").(string)
 	folderId := d.Get("parent_folder_id").(string)
