diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/iam.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/iam.go
index d2ee47880..b48113048 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/iam.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/iam.go
@@ -422,10 +422,13 @@ func jsonPolicyDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 }
 
 func compareIamPolicies(a, b *cloudresourcemanager.Policy) bool {
-	if a.Etag != b.Etag {
-		log.Printf("[DEBUG] policies etag differ: %q vs %q", a.Etag, b.Etag)
-		return false
-	}
+	// There is really no need to compare etags to determine diffs on client side.
+	// It should be passed along to the API for concurrency control, or always
+	// be set to empty string if users want to overwrite the whole policy.
+	//if a.Etag != b.Etag {
+	//	log.Printf("[DEBUG] policies etag differ: %q vs %q", a.Etag, b.Etag)
+	//	return false
+	//}
 	if a.Version != b.Version {
 		log.Printf("[DEBUG] policies version differ: %q vs %q", a.Version, b.Version)
 		return false
