diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgiamresource/resource_iam_audit_config.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgiamresource/resource_iam_audit_config.go
index f4105322e..8d1624e7c 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgiamresource/resource_iam_audit_config.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgiamresource/resource_iam_audit_config.go
@@ -17,6 +17,7 @@ var iamAuditConfigSchema = map[string]*schema.Schema{
 	"service": {
 		Type:        schema.TypeString,
 		Required:    true,
+		ForceNew:    true,
 		Description: `Service which will be enabled for audit logging. The special value allServices covers all services.`,
 	},
 	"audit_log_config": {
