diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go
index 1596bf605..ae6d692f4 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go
@@ -898,6 +898,7 @@ If reconciliation failed, trafficStatuses, observedGeneration, and latestReadyRe
 				Type:        schema.TypeList,
 				Computed:    true,
 				Description: `The Condition of this Service, containing its readiness status, and detailed error information in case it did not reach a serving state. See comments in reconciling for additional information on reconciliation process in Cloud Run.`,
+				MaxItems:    1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"execution_reason": {
