diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go
index ae6d692f4..5f4ba36b2 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2/resource_cloud_run_v2_service.go
@@ -623,16 +623,16 @@ A duration in seconds with up to nine fractional digits, ending with 's'. Exampl
 													Description: `If unspecified, the volume will expose a file whose name is the secret, relative to VolumeMount.mount_path. If specified, the key will be used as the version to fetch from Cloud Secret Manager and the path will be the name of the file exposed in the volume. When items are defined, they must specify a path and a version.`,
 													Elem: &schema.Resource{
 														Schema: map[string]*schema.Schema{
-															"mode": {
-																Type:        schema.TypeInt,
-																Required:    true,
-																Description: `Integer octal mode bits to use on this file, must be a value between 01 and 0777 (octal). If 0 or not set, the Volume's default mode will be used.`,
-															},
 															"path": {
 																Type:        schema.TypeString,
 																Required:    true,
 																Description: `The relative path of the secret in the container.`,
 															},
+															"mode": {
+																Type:        schema.TypeInt,
+																Optional:    true,
+																Description: `Integer octal mode bits to use on this file, must be a value between 01 and 0777 (octal). If 0 or not set, the Volume's default mode will be used.`,
+															},
 															"version": {
 																Type:        schema.TypeString,
 																Optional:    true,
