diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go
index 5cc6c9180..e85e5fcec 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go
@@ -228,18 +228,6 @@ func resourceSecretManagerSecretVersionRead(d *schema.ResourceData, meta interfa
 		return transport_tpg.HandleNotFoundError(err, d, fmt.Sprintf("SecretManagerSecretVersion %q", d.Id()))
 	}
 
-	res, err = resourceSecretManagerSecretVersionDecoder(d, meta, res)
-	if err != nil {
-		return err
-	}
-
-	if res == nil {
-		// Decoding the object has resulted in it being gone. It may be marked deleted
-		log.Printf("[DEBUG] Removing SecretManagerSecretVersion because it no longer exists.")
-		d.SetId("")
-		return nil
-	}
-
 	// Explicitly set virtual fields to default values if unset
 	if _, ok := d.GetOkExists("deletion_policy"); !ok {
 		if err := d.Set("deletion_policy", "DELETE"); err != nil {
