diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go
index 984dd5a17..e733e3cd4 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager/resource_secret_manager_secret_version.go
@@ -200,18 +200,6 @@ func resourceSecretManagerSecretVersionRead(d *schema.ResourceData, meta interfa
 		return transport_tpg.HandleNotFoundError(err, d, fmt.Sprintf("SecretManagerSecretVersion %q", d.Id()))
 	}
 
-	res, err = resourceSecretManagerSecretVersionDecoder(d, meta, res)
-	if err != nil {
-		return err
-	}
-
-	if res == nil {
-		// Decoding the object has resulted in it being gone. It may be marked deleted
-		log.Printf("[DEBUG] Removing SecretManagerSecretVersion because it no longer exists.")
-		d.SetId("")
-		return nil
-	}
-
 	if err := d.Set("enabled", flattenSecretManagerSecretVersionEnabled(res["state"], d, config)); err != nil {
 		return fmt.Errorf("Error reading SecretVersion: %s", err)
 	}
