diff --git a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_service_account.go b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_service_account.go
index bd4337477..15b8162de 100644
--- a/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_service_account.go
+++ b/third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_google_service_account.go
@@ -272,14 +272,15 @@ func resourceGoogleServiceAccountUpdate(d *schema.ResourceData, meta interface{}
 func resourceGoogleServiceAccountImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	config := meta.(*Config)
 	if err := ParseImportId([]string{
+		"projects/(?P<project>[^/]+)/serviceAccounts/(?P<unique_id>[\\d]+)",
 		"projects/(?P<project>[^/]+)/serviceAccounts/(?P<email>[^/]+)",
 		"(?P<project>[^/]+)/(?P<email>[^/]+)",
 		"(?P<email>[^/]+)"}, d, config); err != nil {
 		return nil, err
 	}
 
-	// Replace import id for the resource id
-	id, err := ReplaceVars(d, config, "projects/{{project}}/serviceAccounts/{{email}}")
+	// Replace import id for the resource id, the last portion of the ID can be either the email or unique_id
+	id, err := ReplaceVars(d, config, "projects/{{project}}/serviceAccounts/{{email}}{{unique_id}}")
 	if err != nil {
 		return nil, fmt.Errorf("Error constructing id: %s", err)
 	}
